web/client-api/src/test/java/io/deephaven/web/client/api/widget/plot/ChartDataTestGwt.java

```diff
@@ -89,4 +112,170 @@ public void testModifiedChartData() {
                 })
                 .then(this::finish).catch_(this::report);
     }
+
+    public void testGeneratedChartData() {
+        connect(tickingData)
+                .then(ideSession -> {
+                    delayTestFinish(9870);
+                    Promise<JsTable> inputPromise = ideSession.getTable("input", false);
+                    Promise<JsTable> resultPromise = ideSession.getTable("result", false);
+                    return inputPromise.then(JsTable::inputTable).then(delayFinish(9871)).then(input -> {
+                        delayTestFinish(9872);
+                        return resultPromise.then(result -> {
+                            delayTestFinish(9873);
+                            ChartData chartData = new ChartData(result);
+
+                            // Keep the same function instance, as it is used for caching data
+                            JsFunction<Any, Any> dataFunction = arr -> Js.asAny("::" + arr);
+
+                            BiConsumer<SubscriptionTableData, SubscriptionTableData> consistencyCheck =
+                                    (subscriptionUpdate, snapshot) -> {
+                                        chartData.update(subscriptionUpdate);
+                                        // Verify the "Value" column, the only mutable column, has the same contents
+                                        // regardless of source
+                                        String[] expectedValues = new String[snapshot.getRows().length];
+                                        snapshot.getRows().forEach((row, index) -> {
+                                            expectedValues[index] = row.get(result.findColumn("Value")).asString();
+                                            return null;
+                                        });
+                                        JsArray<Any> values = chartData.getColumn("Value", null, subscriptionUpdate);
+                                        JsArray<Any> valuesWithPrefix =
+                                                chartData.getColumn("Value", dataFunction, subscriptionUpdate);
```

Comment:
I'm not sure this tests the unwrapping bug in the ticket. The first `getColumn` was always fine because it computes if not defined (they're all undefined in the cache at first call). Then after an update w/ modified rows the new rows weren't being unwrapped because `getColumn` doesn't call `func` since the index value is already defined. The modified handler calls the unwrap function in that case

I think a `chartData.getColumn` call is needed right after `chartData` is created to test this case

Reply:
> I think a chartData.getColumn call is needed right after chartData is created to test this case

That is what is happening here, see the test adding `row(3, "F")` to the input table. Then, this same consistency check is run after the client receives the updates.

This consistency check is run after every update, so that we have confidence in all the various kinds of changes that can happen to table data, and that the data afterwards is the same as if would have been if the client had just freshly subscribed.

I verified this by writing this set of tests first, confirming it failed (`F` overwrote `A` instead of `C`), fixed that, confirmed it failed (`F` was read instead of the expected `::F`, fixed that.

Reply:
Ahh I missed the `consistencyCheck =` part of this due to Github word wrapping. Yes this does look right